.github/workflows/test.yml

```diff
@@ -6,6 +6,9 @@ on:
   pull_request:
     branches:
       - main
+  schedule:
+    - cron: "0 0 * * *"
```

Comment:
Figured it'd be good to be running nightly just in case some breaking changes make it in (two PRs in a short time which never get rebased into each other before testing/merging). This file needed a bit of maintenance to make this work - basically, on PRs only changed things get tested, while anything else (workflow dispatch/schedule) runs everything